pkg/tbtc/signing.go

```diff
@@ -2,378 +2,386 @@ package tbtc
-}
+// signBatch performs the signing process for each message from the given
+// messages batch, one after another. If at least one message cannot be signed,
+// this function returns an error. If all messages were signed successfully,
+// a slice of signatures is returned. Order of the returned signatures matches
+// the order of the messages in the batch, i.e. the first signature corresponds
+// to the first message, and so on.
+func (se *signingExecutor) signBatch(
+	ctx context.Context,
+	messages []*big.Int,
+	startBlock uint64,
+) ([]*tecdsa.Signature, error) {
+	wallet := se.wallet()
 
-// signingAttemptFn represents a function performing a signing attempt.
-type signingAttemptFn func(*signingAttemptParams) (*signing.Result, error)
+	walletPublicKeyBytes, err := marshalPublicKey(wallet.publicKey)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal wallet public key: [%v]", err)
+	}
 
-// start begins the signing retry loop using the given signing attempt function.
-// The retry loop terminates when the signing result is produced or the ctx
-// parameter is done, whatever comes first.
-func (srl *signingRetryLoop) start(
-	ctx context.Context,
-	waitForBlockFn waitForBlockFn,
-	signingAttemptFn signingAttemptFn,
-) (*signing.Result, error) {
-	for {
-		srl.attemptCounter++
-
-		// In order to start attempts >1 in the right place, we need to
-		// determine how many blocks were taken by previous attempts. We assume
-		// the worst case that each attempt failed at the end of the signing
-		// protocol.
-		//
-		// That said, we need to increment the previous attempt start
-		// block by the number of blocks equal to the protocol duration and
-		// by some additional delay blocks. We need a small fixed delay in
-		// order to mitigate all corner cases where the actual attempt duration
-		// was slightly longer than the expected duration determined by the
-		// signingAttemptMaxBlockDuration constant.
-		//
-		// For example, the attempt may fail at the end of the protocol but the
-		// error is returned after some time and more blocks than expected are
-		// mined in the meantime.
-		if srl.attemptCounter > 1 {
-			srl.attemptStartBlock = srl.attemptStartBlock +
-				srl.announcementDelayBlocks +
-				srl.announcementActiveBlocks +
-				signingAttemptMaxBlockDuration +
-				srl.attemptDelayBlocks
-		}
+	messagesDigests := make([]string, len(messages))
+	for i, message := range messages {
+		bytes := message.Bytes()
 
-		announcementStartBlock := srl.attemptStartBlock + srl.announcementDelayBlocks
-		err := waitForBlockFn(ctx, announcementStartBlock)
-		if err != nil {
-			return nil, fmt.Errorf(
-				"failed waiting for announcement start block [%v] "+
-					"for attempt [%v]: [%v]",
-				announcementStartBlock,
-				srl.attemptCounter,
-				err,
+		var messageDigest string
+		if len(bytes) > 8 {
+			messageDigest = fmt.Sprintf(
+				"0x%x...%x",
+				bytes[:2],
+				bytes[len(bytes)-2:],
 			)
```

Comment:
Not critical but it's the same pattern [we already discussed](https://github.com/keep-network/keep-core/pull/3404#discussion_r1029257543). Could be worth introducing some utility formatter for shortening digests for logs.

Reply:
Honestly, I'm not sure if there is sense to do it now. Actually, that pattern will be used just here as the `tbtc.go`'s one is temporary and will be removed soon. Not sure if we can write a formatter that will satisfy all possible cases based on our single existing case. However, we should definitely think about it if we realize that pattern becomes common.

Reply:
Makes sense!